packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts

```diff
@@ -241,6 +243,14 @@ export function handleDelimiterKeyDownEvent(editor: IEditor, event: KeyDownEvent
                 }
             }
         }
+    } else {
+        const entity = getSelectedEntity(selection);
+        if (entity && isNodeOfType(entity, 'ELEMENT_NODE') && helper.isNodeInEditor(entity)) {
+            selection.range.selectNode(entity);
+            if (isEnter) {
+                triggerEntityEventOnEnter(editor, entity, rawEvent);
```

Comment:
If this is only for ENTER key, check ENTER key first, before searching the entity. That will have better performance

Reply:
The entity is being searched to be selected when the range is expanded, then when we pressed any key, the key pressed can be inserted in the range of the entity. This is was being handle by `EnterBeforeReadonlyEntityFeature`, but was not handle by the EntityPlugin.

Reply:
hmm. ok. Is this moved from some old code? If so, where is the old code?

Reply:
it is not moved from old code, but here is the code to remove the expanded selection 
![Screenshot 2024-03-19 at 14 21 58](https://github.com/microsoft/roosterjs/assets/87443959/583d5ee3-f721-4141-b95a-3e02d883c5d4)

Reply:
Maybe we should try only delete the selection content and do not prevent default behavior. I'll try this.